protocols/imap/src/test/java/org/apache/james/imap/processor/SetAnnotationProcessorTest.java

```diff
@@ -0,0 +1,183 @@
+
+    private ImapProcessor mockNextProcessor;
+    private MailboxManager mockMailboxManager;
+    private StatusResponseFactory mockStatusResponseFactory;
+    private ImapProcessor.Responder mockResponder;
+    private ImapSession mockImapSession;
+    private MailboxSession mockMailboxSession;
+    
+    private final List<MailboxAnnotation> MAILBOX_ANNOTATIONS = Lists.newArrayList(MailboxAnnotation.newInstance("/private/key", "anyValue"));
+    private final ImapCommand imapCommand = ImapCommand.anyStateCommand("Name");
+    private MailboxSession.User user = mock(MailboxSession.User.class);
+    private StatusResponse okResponse = mock(StatusResponse.class);
+
+    private SetAnnotationRequest request = new SetAnnotationRequest(TAG, imapCommand, ImapConstants.INBOX_NAME, MAILBOX_ANNOTATIONS);
+
+    @Before
+    public void setUp() {
+        setupMockData();
+        processor = new SetAnnotationProcessor(mockNextProcessor, mockMailboxManager, mockStatusResponseFactory);
+        verify(mockMailboxManager).hasCapability(eq(MailboxCapabilities.Annotation));
+    }
+
+    @Test
+    public void getImplementedCapabilitiesShouldContainSupportAnnotationWhenMailboxManagerHasAnnotationCapability() {
+        assertThat(processor.getImplementedCapabilities(null)).containsExactly(ImapConstants.SUPPORTS_ANNOTATION);
+    }
+
+    @Test
+    public void getImplementedCapabilitiesShouldNotContainSupportAnnotationWhenMailboxManagerHasNotAnnotationCapability() {
+        when(mockMailboxManager.hasCapability(eq(MailboxCapabilities.Annotation))).thenReturn(false);
+        assertThat(new SetAnnotationProcessor(mockNextProcessor, mockMailboxManager, mockStatusResponseFactory).getImplementedCapabilities(null)).isEmpty();
+    }
+    
+    @Test
+    public void processShouldProcessRequestAndCheckMailboxExist() throws Exception {
+        MailboxPath inbox = MailboxPath.inbox(mockMailboxSession);
+
+        when(mockMailboxManager.mailboxExists(inbox, mockMailboxSession)).thenReturn(false);
+
+        processor.process(request, mockResponder, mockImapSession);
+
+        verify(mockMailboxManager).startProcessingRequest(mockMailboxSession);
+        verify(mockMailboxManager).endProcessingRequest(mockMailboxSession);
+        verify(mockMailboxManager).mailboxExists(inbox, mockMailboxSession);
+    }
+
+    @Test
+    public void processShouldResponseNoWithFailureMailboxWhenMailboxDoesNotExist() throws Exception {
+        MailboxPath inbox = MailboxPath.inbox(mockMailboxSession);
+
+        when(mockMailboxManager.mailboxExists(inbox, mockMailboxSession)).thenReturn(false);
+
+        processor.process(request, mockResponder, mockImapSession);
+
+        verify(mockMailboxManager, never()).updateAnnotations(any(MailboxPath.class), any(MailboxSession.class), eq(Lists.newArrayList(MAILBOX_ANNOTATIONS)));
+        
+        ArgumentCaptor<HumanReadableText> captorMessage = ArgumentCaptor.forClass(HumanReadableText.class);
```

Comment:
Fields too for your arguments captors. By the way, thanks, I did not know this...